src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,29 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int EIGHTEEN_YEARS_OLD = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new NullPointerException("User can't be Null");
+        }
+        if (Storage.people.contains(user)) {
+            throw new RuntimeException("User already exist");
+        }
+        if (user.getAge() < EIGHTEEN_YEARS_OLD) {
+            throw new RuntimeException("User to young");
+        }
+        if (user.getPassword().length() < 6) {
```

Comment:
It is better to make "6" constant

Reply:
also if password will be null you'll get NullPointerException, because of calling method in null value